From beb85146cd91de37ae455eccb6ab67c393e6e290 Mon Sep 17 00:00:00 2001
From: Matthias Brugger <mbrugger@suse.com>
Date: Fri, 16 Oct 2020 13:20:09 +0200
Subject: [PATCH] aarch64: Fix aarch64 support

When running virtme natively on a aarch64 host, we should use the same
version of GIC as the host. We also don't need to emulate a CPU. Instead
we can run on the host one.

Signed-off-by: Matthias Brugger <mbrugger@suse.com>
---
 virtme/architectures.py | 15 ++++++++++-----
 1 file changed, 10 insertions(+), 5 deletions(-)

diff --git a/virtme/architectures.py b/virtme/architectures.py
index c284416..ba16138 100644
--- a/virtme/architectures.py
+++ b/virtme/architectures.py
@@ -165,12 +165,17 @@ class Arch_aarch64(Arch):
     def qemuargs(is_native):
         ret = Arch.qemuargs(is_native)
 
-        # Emulate a fully virtual system.
-        ret.extend(['-M', 'virt'])
+        if is_native:
+            ret.extend(['-M', 'virt,gic-version=host'])
+            ret.extend(['-cpu', 'host'])
+        else:
+            # Emulate a fully virtual system.
+            ret.extend(['-M', 'virt'])
+
+            # Despite being called qemu-system-aarch64, QEMU defaults to
+            # emulating a 32-bit CPU.  Override it.
+            ret.extend(['-cpu', 'cortex-a57'])
 
-        # Despite being called qemu-system-aarch64, QEMU defaults to
-        # emulating a 32-bit CPU.  Override it.
-        ret.extend(['-cpu', 'cortex-a57'])
 
         return ret
 
-- 
2.28.0

